include/seqan3/alphabet/structure/wuss.hpp

```diff
@@ -0,0 +1,408 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \ingroup structure
+ * \author Joerg Winkler <j.winkler AT fu-berlin.de>
+ * \brief Contains the WUSS format for RNA structure.
+ */
+
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/structure/concept.hpp>
+
+// ------------------------------------------------------------------
+// wuss
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+/*!\brief The WUSS structure alphabet of the characters ".<>_,:~()[]{}AaBbCcDd...Ss".
+ * \ingroup structure
+ *
+ * \details
+ * The symbols ._,:~ denote unpaired characters, brackets <>()[]{} represent base pair interactions and AaBb...Ss
+ * form pseudoknots in the structure. The default alphabet has size 51 (letters until Ss). The size can be varied with
+ * the optional template parameter between 13 (no letters for pseudoknots) and 65 (all Aa-Zz for pseudoknots).
+ *
+ *~~~~~~~~~~~~~~~{.cpp}
+ * <<<___>>>,,<<<__>>>
+ * <<<<_AAAA____>>>>aaaa
+ *~~~~~~~~~~~~~~~
+ */
+
+template <uint8_t SIZE = 51>
+struct wuss
+{
+
+    //!\brief The type of the alphabet when converted to char (e.g. via to_char()).
+    using char_type = char;
+    //!\brief The type of the alphabet when represented as a number (e.g. via to_rank()).
+    using rank_type = uint8_t;
+
+    /*!\name Letter values
+     * \brief Static member "letters" that can be assigned to the alphabet or used in aggregate initialization.
+     * \details Similar to an Enum interface. *Don't worry about the `internal_type`.*
+     */
+    //!\{
+    static const wuss NP; // not paired .
+    static const wuss BL; // bracket left <
+    static const wuss BR; // bracket right >
+    static const wuss NP1; // not paired :
+    static const wuss NP2; // not paired ,
+    static const wuss NP3; // not paired _
+    static const wuss NP4; // not paired ~
+    static const wuss BL1; // bracket left (
+    static const wuss BR1; // bracket right )
+    static const wuss BL2; // bracket left [
+    static const wuss BR2; // bracket right ]
+    static const wuss BL3; // bracket left {
+    static const wuss BR3; // bracket right }
+    // pseudoknots not accessible
```

Comment:
The "enum-like" interface is just for convenience anyway, it's ok, if it doesn't cover all options